homeassistant/components/tedee/__init__.py

```diff
@@ -37,6 +51,56 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
     hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator
 
+    register_lock = asyncio.Lock()
```

Comment:
No need for the lock since you are not actively adding and removing (Withings uses this for the cloud)